coalib/misc/Annotations.py

```diff
@@ -0,0 +1,44 @@
+
+
+def typechain(*args):
+    """
+    Returns function which applies the first transformation it can from args
+    and returns transformed value, raises ValueError when cannot apply none.
+
+    >>> function = typechain(int, ord, None)
+    >>> function("10")
+    10
+    >>> function("a")
+    97
+    >>> function(int)
+    <class 'int'>
+    >>> function(None) is None
+    True
+    >>> function("str")
+    Traceback (most recent call last):
+        ...
+    ValueError: Couldn't convert to any specified type value 'str'.
+
+    :raises TypeError:  Raises when either no functions are specified for
+                        checking or one or more object isn't callable and not
+                        None.
+    """
+    if len(args) == 0:
+        raise TypeError("No arguments were provided.")
+    if not all(arg is None or getattr(arg, '__call__', False) for arg in args):
+        raise TypeError("One of the provided argument is not callable.")
```

Comment:
not sure I agree, say I want to allow 3, 4 and None I'd probably do something like `typechain(3, 4, None)` right? Why limit the type matching to None?